### PR TITLE
Better options handling

### PR DESCRIPTION
**Pull request by OskarStark**
Replaces
* #37

### CODE REVIEW

File: examples/chat-gpt-openai.php [hunk 1]
```diff
@@ -11,13 +11,18 @@
 require_once dirname(__DIR__).'/vendor/autoload.php';
 
 $platform = new OpenAI(HttpClient::create(), getenv('OPENAI_API_KEY'));
-$llm = new Gpt($platform, Version::gpt4oMini());
+$llm = new Gpt($platform, Version::gpt4oMini(), [
+    'temperature' => 0.5,
+    'top_p' => 0.9,
```
Comment (chr-hertel) [inline examples/chat-gpt-openai.php:16]: as example misleading since OpenAI discourages using both in one request

Comment (OskarStark) [inline examples/chat-gpt-openai.php:16]:
```suggestion
```

File: examples/chat-gpt-openai.php [hunk 2]
```diff
@@ -11,13 +11,17 @@
 require_once dirname(__DIR__).'/vendor/autoload.php';
 
 $platform = new OpenAI(HttpClient::create(), getenv('OPENAI_API_KEY'));
-$llm = new Gpt($platform, Version::gpt4oMini());
+$llm = new Gpt($platform, Version::gpt4oMini(), [
+    'temperature' => 0.5,
```
Comment (OskarStark) [inline examples/chat-gpt-openai.php:15]:
```suggestion
    'temperature' => 0.5, // default options for the model
```

File: examples/chat-gpt-openai.php [hunk 3]
```diff
@@ -11,13 +11,17 @@
...
+]);
 
 $chain = new Chain($llm);
 $messages = new MessageBag(
     Message::forSystem('You are a pirate and you write funny.'),
     Message::ofUser('What is the Symfony framework?'),
 );
-$response = $chain->call($messages);
+$response = $chain->call($messages, [
+    'max_tokens' => 500,
```
Comment (OskarStark) [inline examples/chat-gpt-openai.php:24]:
```suggestion
    'max_tokens' => 500, // specific options just for this call
```

**Review by chr-hertel — APPROVED** [review]

**Pull request merged**